### DISCUSSION TITLE
Question about speed of data transfering...

### DISCUSSION THREAD

**Discussion by Yang-Xijie** [insidegui/MultipeerKit#19] [Q&A] [answered]
I wonder what kind of service this framework could cover depending on the transfering speed.

I noticed that in the example app `RemoteCamera`, photos are sent quickly between different divices.

Will it be possible to share video data using this framework? Has someone ever tested the transfering performance?

**Comment by insidegui** [comment]
The performance will vary depending on quite a large number of factors, including but not limited to:

- The path that MultipeerConnectivity is using for the communication (it could be doing peer-to-peer WiFi over AWDL or using the wireless AP)
- The devices involved:
    - The particular model of WiFi chip they have
    - The version of the baseband
    - OS version
- The environment:
    - Is the device currently connected to WiFi. If so, is it using the 2ghz or the 5ghz band
    - How much WiFi traffic is going on to/from the device or in its vicinity
    - How far apart the devices are during transmission

As you can see, a large number of variables can affect the data transfer performance that you should expect. In general, since the underlying transport is pretty much the same, you should expect performance similar to what you get with AirDrop. In ideal conditions (recent devices fairly close to one another) it should be quite fast, probably fast enough to transfer large amounts of data.

In your example of sharing video, I'm not sure if you're referring to streaming video or sending video files. If it's sending video files, you can probably transfer files of significant sizes (a few GB) fairly quickly in ideal conditions. If it's streaming video, then you'd probably be better off by using the stream support built-in to MultipeerConnectivity.

Regardless of your particular use case, MultipeerKit's API was designed with the goal of transferring small amounts of `Codable` structs of well-defined types over the air. As of yet, it doesn't expose MultipeerConnectivity's support for transferring resources (files) or opening real-time streams between devices, which I think would be a better fit for your use case. It wouldn't be too difficult to use MultipeerConnectivity directly though, since [it's well documented here](https://developer.apple.com/documentation/multipeerconnectivity).

**Reply by Yang-Xijie** [reply]
Thanks for your reply! 

What I want to achieve is the function like video streaming. I will try it using the original framework.

BTW, `MultipeerKit` is really handy to use 😃!

**Comment by Yang-Xijie** [comment, marked as answer]
I roughly tested the speed of the stream (not professionally), the results are as follows:

Regardless of IO processing, speed reaches **20 MB/s** around between a simulator on Macbook Pro and an iPad Pro (connected to the same AP). When closing the wifi, speed reaches 12MB/s. It has the ability to support many services.

Here is the demo app <https://github.com/Yang-Xijie/ExPeerMessage>.

However, I also tried Airdrop, which reached **60MB/s** around. It might be the problem of the method I used to transfer test data. Or I guess that Airdrop uses some other technology.

**Reply by insidegui** [reply]
Thanks for reporting the results!

AirDrop uses AWDL just like MC does, but I guess since MC is working at a higher level of abstraction, there’s other stuff it does behind the scenes that might be impacting performance. 

I won’t have the time to check out your example today, but something to try out would be to ensure that you stop advertising and browsing after the connection has been established, since advertising and browsing can definitely impact WiFi performance.
